var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -192,8 +195,7 @@ def determine_bootstrap_options(self, spec, withLibs, options):
         options.append("--with-libraries=%s" % ','.join(withLibs))
 
         if '+python' in spec:
-            options.append('--with-python=%s' %
-                           join_path(spec['python'].prefix.bin, 'python'))
+            options.append('--with-python=%s' % python_exe)
```

Comment:
Exactly. I need to use the same functionality in #3210 too, so it's a common need.